sniffer.cpp

```diff
@@ -0,0 +1,199 @@
+#include <arpa/inet.h>
+#include <getopt.h>
+#include <netinet/in.h>
+#include <netinet/ip.h>
+#include <netinet/ip_icmp.h>
+#include <netinet/tcp.h>
+#include <netinet/udp.h>
+#include <pcap.h>
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+#include <string>
+#include <chrono>
+#include <deque>
+#include <map>
+
+using namespace std;
+
+const int THRESHOLD = 10; // requests
+const int PERIOD = 5; // seconds
+
+std::chrono::high_resolution_clock::time_point initial_time;
```

Comment:
Not sure about that type :(

Reply:
It's fine even though a monotonic clock would've been better. And we don't necessarily need high precision since you are only using seconds.